presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -215,7 +215,7 @@ public Expression visitWindow(WindowNode node, Void context)
     @Override
     public Expression visitUnion(UnionNode node, Void context)
     {
-        return deriveCommonPredicates(node, source -> node.outputSymbolMap(source).entries());
+        return deriveCommonPredicates(node, source -> node.getMultiSourceSymbolMapping().outputSymbolMap(source).entries());
```

Comment:
> The same thing exists in ExchangeNode, I will try to use MultiSourceSymbolMapping there as a follow up of this.

cmt msg doesn't need to contain information about your plans for the future. Consider removing or rephrasing as "`MultiSourceSymbolMapping` can later also be used in `ExchangeNode`."